### PR TITLE
DB-10526: fix CachedOperation initialization

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Review by msirek — APPROVED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java [hunk 1]
```diff
@@ -110,6 +110,11 @@ public String prettyPrint(int indentLevel) {
         if (!isOpen)
             throw new IllegalStateException("Operation is not open");
         populateCache();
+        try {
+            source.init(SpliceOperationContext.newContext(activation));
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java:114]: @jyuanca shouldn't we have called source.init() during the initialization of the CachedOperation? Why do we still need to init the source again? Thanks!

Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java:114]: populateCache() closes source, so it needs to be re-initalized and open again.

**Review by yxia92 — APPROVED** [review]
+1

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java (hunk 1 above)
Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/CachedOperation.java:114]: Thanks @jyuanca !

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**